### ISSUE TITLE
Print the number of trainable parameters

### ISSUE THREAD

**Issue by pescap** [lululxvi/deepxde#498]
Hi, 
Do you consider adding a Keras-inspired model.summary() for DeepXDE?
Is there a simple way to access the number of trainable parameters for a model?

Paul

**Comment by pescap** [comment]
#https://github.com/lululxvi/deepxde/issues/292

**Comment by lululxvi** [comment]
`model.summary()` is definitely useful, but it may not be easy to do, unless the underlying model is a pure Keras model.

For TF 1, you can use
https://github.com/lululxvi/deeponet/blob/7a110731b73bcdeabfcff819f353fa87b0543cff/src/deeponet_pde.py#L173
For other backends, there should exist similar functions.

**Comment by pescap** [comment]
Hi! It could be interesting to use "Enhancement" labels for this kind of issues. 
Paul

**Comment by lululxvi** [comment]
Good suggestion. The issue is that I don't have enough time to track these "Enhancement" labels.

**Comment by pescap** [comment]
Hi, 
For fully-connected neural network, the number of trainable parameters is with an analytic formula:

```
Input dimension: d
L-layer NN with N neurons per layer for 1\leq l \leq L-1
Output dimension: r
```
The formula is:
```
|\theta|  = N (d+1) + (L-2) N (N+1) + r (N+1)
```

I propose to: 

- [ ] Define `Model.total_trainable_parameters` 
- [ ] Explain how to obtain the formula somewhere in `docs`

Questions: 

1. Where shall I add the function? Whithin `nn.fnn` for each backend and transfer its value to model if defined? Or somewhere in `nn`, as the formula is backend-independent?
2. In which section shall I add the formula proof?

**Comment by pescap** [comment]
> The formula is only for NN with the same width for each layer.

I'll define `num_trainable_parameters` for general NN with widths `N_0, \cdots, N_L`. 

> * If you use TF v1 backend, then the code above can show the number of parameters

Yes, as 

```print("# Parameters:", np.sum([np.prod(v.get_shape().as_list()) for v in tf.trainable_variables()]))```

works once one runs `model.compile()`, and not directly for the `net`. 

> * For backend TF v2 and PyTorch, it should be even easier. TF v2 uses `tf.keras` and PyTorch uses `torch.nn`
> 

I'll test and compare both solutions (the formula and the `keras`/`torch`-based version). 

> I think each nn can have a function to return the number of trainable weights
> 
> ```
> net = dde.nn.FNN(...)
> net.num_trainable_parameters()
> ```
